test/grpc/integration/server_test.exs

```diff
@@ -157,4 +162,20 @@ defmodule GRPC.Integration.ServerTest do
       end)
     end)
   end
+
+  test "headers set on channel are present in receiving server" do
+    run_server([HelloServer], fn port ->
+      token = "Bearer TOKEN"
+
+      {:ok, channel} =
+        GRPC.Stub.connect("localhost:#{port}",
+          headers: [{"authorization", token}]
+        )
+
+      {:ok, reply} =
+        channel |> Helloworld.Greeter.Stub.check_headers(Helloworld.HeaderRequest.new())
+
+      assert reply.authorization == token
```

Comment:
Verify roundtrip.

My understanding is that headers must be set on client and if desired set anew on server for the return trip. So I return them here in the `Reply` for testing.